daphne/src/testing.rs

```diff
@@ -470,32 +474,156 @@ impl DapLeader<BearerToken> for MockAggregator {
         Ok(HashMap::from([(selector.task_id.clone(), reports)]))
     }
 
-    async fn init_collect_job(&self, _collect_req: &CollectReq) -> Result<Url, DapError> {
-        unreachable!("not implemented");
+    // Called after receiving a CollectReq from Collector.
+    async fn init_collect_job(&self, collect_req: &CollectReq) -> Result<Url, DapError> {
+        let mut rng = thread_rng();
+        let task_config = self
+            .get_task_config_for(&collect_req.task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        let mut leader_state_store_mutex_guard = self
+            .leader_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let leader_state_store = leader_state_store_mutex_guard.deref_mut();
+
+        // Store Collect ID and CollectReq into LeaderState.
+        let collect_id = Id(rng.gen());
+        if !leader_state_store.contains_key(&collect_req.task_id) {
+            leader_state_store.insert(collect_req.task_id.clone(), LeaderState::new());
+        }
```

Comment:
It should be possible to do this more efficiently with https://doc.rust-lang.org/stable/std/collections/hash_map/enum.Entry.html#method.or_insert_with.

Reply:
Great suggestion, I've replaced the original code with `or_insert_with`.